integration-tests/hazelcast-client/src/test/java/io/quarkus/it/hazelcast/client/HazelcastServerTestResource.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.it.hazelcast.client;
+
+import java.util.Collections;
+import java.util.Map;
+
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+
+public class HazelcastServerTestResource implements QuarkusTestResourceLifecycleManager {
+
+    private volatile HazelcastInstance member1;
+
+    @Override
+    public Map<String, String> start() {
+        member1 = Hazelcast.newHazelcastInstance();
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public void stop() {
+        Hazelcast.shutdownAll();
```

Comment:
Maybe call member1.shutdown() just in case?

Reply:
All the client instances get registered internally so I'd say this is the "just in case" option